INMOOV-Head-Control/StandardServo.cpp

```diff
@@ -10,17 +10,44 @@ StandardServo::StandardServo(byte pin, float PWMPerDegree, int minRotation, int
   m_PWMPerDegree = PWMPerDegree;
   m_minRotation = ((minRotation * m_PWMPerDegree) + 1000);
   m_maxRotation = ((maxRotation * m_PWMPerDegree) + 1000);
-  m_defaultRotation  = ((defaultRotation * m_PWMPerDegree) + 1000);
+  m_reverseDirection = reverseDirection;
+  m_defaultRotation  = constrain(((defaultRotation * m_PWMPerDegree) + 1000), m_minRotation, m_maxRotation);
 }
 
 void StandardServo::attach() {
   m_servo.attach(m_pin);
-  m_servo.writeMicroseconds(m_defaultRotation);
+  moveToTargetMicroseconds(m_defaultRotation);
   m_currentRotation = m_defaultRotation;
 }
 
-void StandardServo::moveTowardsTarget(float degreesToMove) {
-  m_currentRotation += (degreesToMove * m_PWMPerDegree);
-  Serial.println(m_currentRotation);
-  m_servo.writeMicroseconds(m_currentRotation);
+bool StandardServo::moveTowardsTarget(float degreesToMove) {
+  int newRotation = m_currentRotation + (degreesToMove * m_PWMPerDegree);//Get the new roation
+  return moveToTargetMicroseconds(newRotation);
 }
+
+bool StandardServo::moveToTarget(int target){
+  target = ((target * m_PWMPerDegree) + 1000);
+  return moveToTargetMicroseconds(target);
+}
+
+bool StandardServo::moveToTargetMicroseconds(int target){
+  bool validMove;
+  if((target > m_maxRotation) || (target < m_minRotation)){//Check if it is in bounds
+    validMove = false;
+  }else{
+    validMove = true;
+  }
+  target = constrain(target, m_minRotation, m_maxRotation);//Constain target roation to bounds
+  if(m_reverseDirection){
+    m_servo.writeMicroseconds(map(target,1000,2000,2000,1000));//Write target rotation flipped to servo
```

Comment:
use m_minPWM & m_maxPWM in the map() function

Reply:
Will do